packages/odyssey-react-mui/scripts/properties-to-ts.ts

```diff
@@ -41,55 +42,71 @@ const convert = (baseFiles, propertiesTargetDir) => {
         propertiesTargetDir,
         filename.replace(extension, ".ts")
       );
+
       writeFileSync(
         targetFile,
-        `export const translation = ${JSON.stringify(obj)};`
+        `export const translation = ${JSON.stringify(propertiesJson)};`
       );
     });
-  }
+  });
 };
 
-async function convertPropertiesToJson({ resourcePath, targetJsonPath }) {
+async function convertPropertiesToJson({
+  resourcePath,
+  targetJsonPath,
+}: {
+  resourcePath: string;
+  targetJsonPath: string;
+}) {
   const sourceDirectory = resolve(resourcePath);
   const propertiesTargetDirectory = resolve(targetJsonPath);
 
   if (!existsSync(sourceDirectory)) {
     mkdirSync(sourceDirectory);
   }
+
   if (existsSync(propertiesTargetDirectory)) {
     rmSync(propertiesTargetDirectory, { recursive: true, force: true });
   }
+
   mkdirSync(propertiesTargetDirectory);
 
-  let baseFiles = await readdir(sourceDirectory);
-  convert(baseFiles, propertiesTargetDirectory);
+  const propertiesFilePaths = await readdir(sourceDirectory, {
+    recursive: true,
+  });
+
+  convert(
+    propertiesFilePaths
+      .filter((propertiesFilePath) =>
+        propertiesFilePath.endsWith(".properties")
+      )
+      .map((propertiesFilePath) => join(sourceDirectory, propertiesFilePath)),
+    propertiesTargetDirectory
+  );
 }
 
 yargs(hideBin(process.argv))
   .scriptName("properties-to-ts")
   .usage("$0 <cmd> [args]")
   .command(
-    "bundle",
-    "Converts properties file to ts",
-    (yargs) => {
+    "bundle [resourcePath] [targetJsonPath]",
+    "Converts `properties` files to TypeScript types.",
+    (yargs) =>
       yargs
         .positional("resourcePath", {
-          type: "string",
           default: "src/properties",
           describe: "A relative path to resources based on cwd.",
+          type: "string",
         })
         .positional("targetJsonPath", {
-          type: "string",
           default: "src/properties/ts",
           describe: "A relative path to directory for ts file output",
-        });
-    },
-    function (argv) {
-      const { resourcePath, targetJsonPath } = argv;
-
+          type: "string",
+        }),
+    (argv) => {
       convertPropertiesToJson({
-        resourcePath,
-        targetJsonPath,
+        resourcePath: argv.resourcePath,
```

Comment:
can we just do:
```
convertPropertiesToJson({ ...argv });

```

instead? or is that too vague haha

Reply:
I'd prefer to be explicit just so it's easier for the next dev.